packages/guides-restructured-text/src/RestructuredText/Parser/InterlinkParser.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\RestructuredText\Parser;
+
+interface InterlinkParser
+{
+    /** @return array{interlink:?string,reference:string} */
+    public function extractInterlink(string $fullReference): array;
```

Comment:
Please use a class instead of a typed array.  `Interlink` would be a good candidate name.